x-pack/plugins/alerting/server/routes/lib/count_usage_of_predefined_ids.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UsageCounter } from 'src/plugins/usage_collection/server';
+
+export interface CountUsageOfPredefinedIdsOptions {
+  predefinedId?: string;
+  spaceId?: string;
+  usageCounter?: UsageCounter;
+}
+
+export function countUsageOfPredefinedIds({
+  predefinedId,
+  spaceId,
+  usageCounter,
+}: CountUsageOfPredefinedIdsOptions) {
+  if (predefinedId) {
+    if (usageCounter) {
+      const usageCounterName =
+        spaceId === undefined || spaceId === 'default'
+          ? 'ruleCreatedWithPredefinedIdInDefaultSpace'
+          : 'ruleCreatedWithPredefinedIdInCustomSpace';
```

Comment:
That's a good idea! I'm currently trying to figure out how to get access to the telemetry cluster to see how these daily events are visualized, but given that it's an array of objects, it's likely that it might not be straightforward to do a sum. Updated [in this commit](https://github.com/elastic/kibana/pull/108572/commits/6859712a117255bf18db15c01681bb1be890c8cb) with your suggestion.

Reply:
>but given that it's an array of objects, it's likely that it might not be straightforward to do a sum.

Oh, interesting 🤔 at least we'll start by tracking and then we can figure out visualizing if ever it's a problem.

I'll reach out with the telemetry cluster info.

Reply:
One thing to note is the telemetry team is able to perform transformations on the data in the "middle layer" of the process (which is basically a cloud service that sits in the middle of Kibana sending the data, and it being indexed into the ES cluster) so it's possible to fix any issues outside of our deployment model

Reply:
That's really good to know! Thanks, I'll add it to my list of questions (w/ the increasing field count each release).

Reply:
That is interesting! Are those transformations documented anywhere?

Reply:
@ymao1 @mikecote The transformations are part of the new telemetry system we've been building out 😀 

We did a [presentation at GAH](https://drive.google.com/file/d/1akglkIuoHTq1W-MUc4Lko67zv4xzU074/view?usp=sharing) about the capabilities, and we've got docs on how to build/use a custom index to take advantage of them [here](https://github.com/elastic/infra/blob/master/docs/telemetry/custom-index-creation.md). The goal is to make things as self-service as possible, so folks like yourselves can have full control over how the data gets indexed.

Reply:
@mindbat Thank you for the links! We will review them and look into how we can take advantage of this system!